### PR TITLE
Make sure DataOutFaces also works in parallel.

### LINKED ISSUE #2053: DataOutFaces doesn't play nice with parallel::distributed::Triangulation
See attached testcase by Heikki Virtanen <heikki.a.virtanen@hotmail.com>.

The error message is something of the sort
```
--------------------------------------------------------
An error occurred in line <94> of file </u/bangerth/p/deal.II/2/dealii/source/numerics/data_out_faces.cc> in function
    void dealii::DataOutFaces<dim, DoFHandlerType>::build_one_patch(const FaceDescriptor*, dealii::internal::DataOutFaces::ParallelData<dealii::DataOutFaces<dim, DoFHandlerType>::dimension, dealii::DataOutFaces<dim, DoFHandlerType>::dimension>&, dealii::DataOutBase::Patch<(dealii::DataOutFaces<dim, DoFHandlerType>::dimension - 1), dealii::DataOutFaces<dim, DoFHandlerType>::space_dimension>&) [with int dim = 2, DoFHandlerType = dealii::DoFHandler<2>, dealii::DataOutFaces<dim, DoFHandlerType>::FaceDescriptor = std::pair<dealii::TriaIterator<dealii::CellAccessor<2, 2> >, unsigned int>, typename dealii::DataOut_DoFData<DoFHandlerType, (dealii::DataOutFaces<dim, DoFHandlerType>::dimension - 1), dealii::DataOutFaces<dim, DoFHandlerType>::dimension>::cell_iterator = dealii::TriaIterator<dealii::CellAccessor<2, 2> >]
The violated condition was: 
    cell_and_face->first->is_locally_owned()
The name and call sequence of the exception was:
    ExcNotImplemented()
Additional Information: 
You are trying to use functionality in deal.II that is currently not implemented. In many cases, this indicates that there simply didn't appear much of a need for it, or that the author of the original code did not have the time to implement a particular case. If you hit this exception, it is therefore worth the time to look into the code to find out whether you may be able to implement the missing functionality. If you do, please consider providing a patch to the deal.II development sources (see the deal.II website on how to contribute).
```

[test.txt](https://github.com/dealii/dealii/files/86405/test.txt)

### PR DESCRIPTION
**Pull request by bangerth**
Fixes #2053.

### CODE REVIEW

**Comment by drwells** [comment]
I believe the output file should be renamed to include `with_petsc=true`.

**Comment by bangerth** [comment]
> I believe the output file should be renamed to include |with_petsc=true|.

Good spot. Done.

File: tests/mpi/data_out_faces_01.cc [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+    DataOut<dim> data_out;
+    data_out.attach_dof_handler (dof_handler);
+    data_out.add_data_vector (solution, "u");
+    Vector<float> subdomain (triangulation.n_active_cells());
+    for (unsigned int i=0; i<subdomain.size(); ++i)
+      subdomain(i) = triangulation.locally_owned_subdomain();
+    data_out.add_data_vector (subdomain, "subdomain");
+    // make sure the following works in parallel
+    data_out.build_patches ();
```
Comment (drwells) [inline tests/mpi/data_out_faces_01.cc]: Would you mind getting rid of the `subdomain` part? It is not related to the rest of the test.

**Comment by drwells** [comment]
Up to the minor (and not necessary) point above this looks good. I checked the output with paraview and things seem correct, so feel free to merge with or without (up to you) the `subdomain` part above.

**Comment by bangerth** [comment]
Thanks, @drwells , for double checking! 

I've made the change requested.

**Comment by drwells** [comment]
@bangerth Not a problem.

I have one more question about this: the current test just ensures that everything runs but it does not verify the output. Would `vtk` output from this test be deterministic, or might it vary with different `p4est` versions?

**Comment by bangerth** [comment]
This should be deterministic. Patch in #2059.

**Pull request merged**